conn.go

```diff
@@ -172,7 +172,7 @@ func Connect(addr string, cfg *ConnConfig, errorHandler ConnErrorHandler, sessio
 		r:            bufio.NewReader(conn),
 		cfg:          cfg,
 		uniq:         make(chan int, streams),
-		calls:        make([]callReq, streams),
+		calls:        make([]*callReq, streams), // TODO(zariel): use a sparse map instead?
```

Comment:
using a map[int]*callReq would make sense now.  Delete the entry from the map when you put it in the sync.Pool to free up memory.  We don't need to hold references to where every possible stream may be help anymore.